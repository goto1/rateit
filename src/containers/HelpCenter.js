import React from "react";
import PropTypes from "prop-types";
import { Formik } from "formik";
import Yup from "yup";
import { isSubmissionDisabled } from "../utils/FormUtils";
import {
  Button,
  ContentBlock,
  ContentBlockTitle,
  InputElement,
  List,
  NavCenter,
  NavLeft,
  Navbar,
  Textarea
} from "../components/f7";
import { Page } from "framework7-react";

export let ContactForm = ({
  errors,
  handleBlur,
  handleChange,
  handleSubmit,
  isSubmitting,
  touched,
  values
}) => {
  const disableSubmission = isSubmissionDisabled({
    isSubmitting,
    errors,
    touched
  });
  return (
    <form onSubmit={handleSubmit}>
      <List inset>
        <InputElement
          icon="info"
          name="title"
          onBlur={handleBlur}
          onChange={handleChange}
          placeholder="Title..."
          type="text"
          valid={errors.title && touched.title && false}
          value={values.title}
        />
        <Textarea
          icon="comment"
          name="message"
          onBlur={handleBlur}
          onChange={handleChange}
          placeholder="Your message here..."
          valid={errors.message && touched.message && false}
          value={values.message}
        />
      </List>
      <ContentBlock>
        <Button
          big
          color="green"
          disabled={disableSubmission}
          fill
          type="submit"
        >
          Submit
        </Button>
      </ContentBlock>
    </form>
  );
};

ContactForm.propTypes = {
  errors: PropTypes.object,
  handleBlur: PropTypes.func,
  handleChange: PropTypes.func,
  handleSubmit: PropTypes.func,
  isSubmitting: PropTypes.bool,
  touched: PropTypes.object,
  values: PropTypes.object
};

ContactForm = Formik({
  mapPropsToValues: props => ({
    title: "",
    message: ""
  }),
  validationSchema: Yup.object().shape({
    title: Yup.string().required(),
    message: Yup.string().required()
  }),
  handleSubmit: (values, { props, setErrors, setSubmitting }) => {
    console.log(values);
  }
})(ContactForm);

export const Content = () =>
  <div>
    <Navbar>
      <NavLeft backLink="Back" sliding />
      <NavCenter sliding>Help Center</NavCenter>
    </Navbar>
    <ContentBlockTitle>Contact form</ContentBlockTitle>
    <ContactForm />
  </div>;

const HelpCenter = () =>
  <Page>
    <Content />
  </Page>;

export default HelpCenter;
