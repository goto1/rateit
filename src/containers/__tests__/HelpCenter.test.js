import React from "react";
import { Content, ContactForm } from "../HelpCenter";
import {
  Button,
  ContentBlockTitle,
  InputElement,
  NavCenter,
  NavLeft,
  Navbar,
  Textarea
} from "../../components/f7";
import renderer from "react-test-renderer";
import { shallow, mount } from "enzyme";

describe("HelpCenter container", () => {
  describe("ContactForm", () => {
    const getContactForm = props => {
      const newProps = {
        errors: {},
        handleBlur: jest.fn(),
        handleChange: jest.fn(),
        handleSubmit: jest.fn(),
        isSubmitting: false,
        touched: {},
        values: { title: "", message: "" },
        ...props
      };
      return <ContactForm {...newProps} />;
    };

    it("should render correctly", () => {
      const ContactForm = getContactForm();
      const tree = renderer.create(ContactForm).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should render all form elements", () => {
      const ContactForm = getContactForm();
      const wrapper = mount(ContactForm);
      const inputExists = wrapper.find(InputElement).exists();
      const textareaExists = wrapper.find(Textarea).exists();
      const buttonExists = wrapper.find(Button).exists();
      const actual = inputExists && textareaExists && buttonExists;
      const expected = true;
      expect(actual).toBe(expected);
    });

    it("should submit the form and disable the submit button", () => {
      const ContactForm = getContactForm();
      const element = mount(ContactForm).find("form");
      element.simulate("submit", {
        preventDefault: () => {}
      });
      const actual = element.find(Button).props().disabled;
      const expected = true;
      expect(actual).toBe(true);
    });
  });

  describe("Content", () => {
    it("should render correctly", () => {
      const tree = renderer.create(<Content />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should render all navbar elements", () => {
      const wrapper = shallow(<Content />);
      const navbarExists = wrapper.find(Navbar).exists();
      const navLeftExists = wrapper.find(NavLeft).exists();
      const navCenterExists = wrapper.find(NavCenter).exists();
      const actual = navbarExists && navLeftExists && navCenterExists;
      const expected = true;
      expect(actual).toBe(expected);
    });

    it("should render a ContactBlockTitle component", () => {
      const wrapper = shallow(<Content />);
      const actual = wrapper.find(ContentBlockTitle).exists();
      const expected = true;
      expect(actual).toBe(expected);
    });

    it("should render a ContactForm component", () => {
      const wrapper = shallow(<Content />);
      const actual = wrapper.find(ContactForm).exists();
      const expected = true;
      expect(actual).toBe(expected);
    });
  });
});
