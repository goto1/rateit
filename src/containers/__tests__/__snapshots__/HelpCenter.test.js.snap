// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpCenter container ContactForm should render correctly 1`] = `
.jwltLt {
  background-color: #369947 !important;
  border-color: #369947 !important;
  color: #F1F1F5 !important;
  opacity: .65;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  -webkit-letter-spacing: .5px;
  -moz-letter-spacing: .5px;
  -ms-letter-spacing: .5px;
  letter-spacing: .5px;
  width: 100%;
}

.fYkctb {
  margin-left: 15px;
}

.fYkctb input {
  color: black;
}

.bqyzQp {
  position: relative;
}

.dgftDG {
  position: absolute !important;
  color: red;
  top: 35%;
  right: 5px;
}

.irilpd {
  position: relative;
}

<form
  onSubmit={[Function]}
>
  <div
    className="list-block inset"
  >
    <ul>
      <div
        className="sc-ifAKCX item-content bqyzQp"
      >
        <div
          className="item-media"
        >
          <i
            className="material-icons icon"
          >
            info
          </i>
        </div>
        <div
          className="item-input sc-htpNat fYkctb"
        >
          <input
            name="title"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Title..."
            type="text"
            value=""
          />
        </div>
      </div>
      <li
        className="align-top"
      >
        <div
          className="sc-gZMcBi item-content irilpd"
        >
          <i
            className="material-icons icon sc-iwsKbI dgftDG"
          >
            warning
          </i>
          <div
            className="item-media"
          >
            <i
              className="material-icons icon"
            >
              comment
            </i>
          </div>
          <div
            className="item-inner"
          >
            <div
              className="item-input"
            >
              <textarea
                name="message"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Your message here..."
                value=""
              />
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="content-block  "
  >
    <button
      className="button button-big sc-bdVaJa jwltLt"
      disabled={true}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`HelpCenter container Content should render correctly 1`] = `
.jwltLt {
  background-color: #369947 !important;
  border-color: #369947 !important;
  color: #F1F1F5 !important;
  opacity: .65;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  -webkit-letter-spacing: .5px;
  -moz-letter-spacing: .5px;
  -ms-letter-spacing: .5px;
  letter-spacing: .5px;
  width: 100%;
}

.cYrpQX {
  margin-top: 25px !important;
}

.fYkctb {
  margin-left: 15px;
}

.fYkctb input {
  color: black;
}

.bqyzQp {
  position: relative;
}

.dgftDG {
  position: absolute !important;
  color: red;
  top: 35%;
  right: 5px;
}

.irilpd {
  position: relative;
}

<div>
  <div
    className="navbar"
  >
    <div
      className="navbar-inner"
    >
      <div
        className="left sliding"
      >
        <a
          className="back link"
        >
          <i
            className="icon icon-back"
          />
          <span>
            Back
          </span>
        </a>
      </div>
      <div
        className="center sliding"
      >
        Help Center
      </div>
    </div>
  </div>
  <div
    className="content-block-title sc-bwzfXH cYrpQX"
  >
    Contact form
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="list-block inset"
    >
      <ul>
        <div
          className="sc-ifAKCX item-content bqyzQp"
        >
          <div
            className="item-media"
          >
            <i
              className="material-icons icon"
            >
              info
            </i>
          </div>
          <div
            className="item-input sc-htpNat fYkctb"
          >
            <input
              name="title"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Title..."
              type="text"
              value=""
            />
          </div>
        </div>
        <li
          className="align-top"
        >
          <div
            className="sc-gZMcBi item-content irilpd"
          >
            <i
              className="material-icons icon sc-iwsKbI dgftDG"
            >
              warning
            </i>
            <div
              className="item-media"
            >
              <i
                className="material-icons icon"
              >
                comment
              </i>
            </div>
            <div
              className="item-inner"
            >
              <div
                className="item-input"
              >
                <textarea
                  name="message"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Your message here..."
                  value=""
                />
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="content-block  "
    >
      <button
        className="button button-big sc-bdVaJa jwltLt"
        disabled={true}
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
